src/openrct2/actions/ChangeMapSizeAction.cpp

```diff
@@ -0,0 +1,80 @@
+/*****************************************************************************
+ * Copyright (c) 2014-2021 OpenRCT2 developers
+ *
+ * For a complete list of all authors, please refer to contributors.md
+ * Interested in contributing? Visit https://github.com/OpenRCT2/OpenRCT2
+ *
+ * OpenRCT2 is licensed under the GNU General Public License version 3.
+ *****************************************************************************/
+
+#include "ChangeMapSizeAction.h"
+
+#include "../Context.h"
+#include "../drawing/IDrawingEngine.h"
+#include "../ui/UiContext.h"
+#include "../ui/WindowManager.h"
+#include "../windows/Intent.h"
+
+ChangeMapSizeAction::ChangeMapSizeAction(const MapSizeChange& sizeChange)
+    : _sizeChange(sizeChange)
+{
+}
+
+uint16_t ChangeMapSizeAction::GetActionFlags() const
+{
+    return GameAction::GetActionFlags() | GameActions::Flags::AllowWhilePaused;
+}
+
+void ChangeMapSizeAction::Serialise(DataSerialiser& stream)
+{
+    GameAction::Serialise(stream);
+    stream << DS_TAG(_sizeChange);
+}
+
+GameActions::Result::Ptr ChangeMapSizeAction::Query() const
+{
+    if (_sizeChange == MapSizeChange::Increase)
+    {
+        if (gMapSize >= MAXIMUM_MAP_SIZE_TECHNICAL)
+        {
+            return std::make_unique<GameActions::Result>(
+                GameActions::Status::InvalidParameters, STR_CANT_INCREASE_MAP_SIZE_ANY_FURTHER, STR_NONE);
+        }
+    }
+    else
+    {
+        if (gMapSize < 16)
+        {
+            return std::make_unique<GameActions::Result>(
+                GameActions::Status::InvalidParameters, STR_CANT_DECREASE_MAP_SIZE_ANY_FURTHER, STR_NONE);
+        }
+    }
+    return std::make_unique<GameActions::Result>();
+}
+
+GameActions::Result::Ptr ChangeMapSizeAction::Execute() const
+{
+    if (_sizeChange == MapSizeChange::Increase)
+    {
+        gMapSize++;
+        map_extend_boundary_surface();
+    }
+    else
+    {
+        gMapSize--;
+        map_remove_out_of_range_elements();
+    }
+
+    auto* ctx = OpenRCT2::GetContext();
+    auto uiContext = ctx->GetUiContext();
+    auto* windowManager = uiContext->GetWindowManager();
+
+    windowManager->BroadcastIntent(Intent(INTENT_ACTION_MAP));
+    gfx_invalidate_screen();
+    return std::make_unique<GameActions::Result>();
+}
+
+void ChangeMapSizeAction::AcceptParameters(GameActionParameterVisitor& visitor)
+{
+    visitor.Visit("sizeChange", _sizeChange);
```

Comment:
How does this get serialised? Does it become, 0 or 1?

Reply:
It is, these are the default values for an enum. I can change it to -1, 1.